### PR TITLE
Create component

### PR DESCRIPTION
**Pull request by hkhattabii**
### ID
2c9fdc0

Enable the users to create components directly in the UI instead of passing a csv file.

### CODE REVIEW

File: src/models/state/ui.models.ts [hunk 1]
```diff
@@ -31,12 +33,15 @@ export interface ITriggerFlashMessagePayload {
     navigateToRoute?: INavigateToRoute;
 }
 
-interface IListFilters<ColumnNames> {
-    filters: ListFilters<Partial<ColumnNames>>;
+interface IListFilters<T> {
+    filters: ListFilters<Partial<T>>;
     page: number;
-    sortedColumn: ISortedColumn<ColumnNames>;
+    sortedColumn: ISortedColumn<T>;
```
Comment (BerrevoetsRobbe) [inline src/models/state/ui.models.ts:39]: why is this changed?

File: src/state/entities/components/triggers.ts [hunk 2]
```diff
@@ -1,12 +1,73 @@
...
+            asyncEntityKey: ASYNC_ENTITY_KEYS.componentDetail,
+        },
+        extraInputSelector: () => payload,
+        notificationsToTrigger: [StateChangeNotification.DESIGN_COMPONENT_DETAIL],
+        onSuccess: ({ dispatch }) => {
+            dispatch(triggerFlashMessage({
+                translationKey: 'flash_messages.component.edit',
+                type: 'success',
+            }));
+        },
```
Comment (BerrevoetsRobbe) [inline src/state/entities/components/triggers.ts:53]: is this also done for scripts?

File: src/views/doc/OpenAPIOverview/index.tsx [hunk 3]
```diff
@@ -63,7 +63,7 @@ interface IComponentState {
     isConnectionEditDialogOpen: boolean;
     isComponentEditDialogOpen: boolean;
     connectionToEdit?: IConnectionEntity | undefined;
-    componentToEdit?: IComponentEntity | undefined;
+    componentToEdit?: IComponent | undefined;
```
Comment (BerrevoetsRobbe) [inline src/views/doc/OpenAPIOverview/index.tsx:66]: isn't it needed to have a dedicated IComponent and IComponentOpenApi ?

**Review by BerrevoetsRobbe — CHANGES_REQUESTED** [review]

File: src/models/state/ui.models.ts (hunk 1 above)
Comment (hkhattabii) [reply src/models/state/ui.models.ts:39]: Because the sorted column was only for "ColumnNames" types which has column names for script only. I created a generic sortedColum to take other column type

File: src/views/doc/OpenAPIOverview/index.tsx (hunk 3 above)
Comment (hkhattabii) [reply src/views/doc/OpenAPIOverview/index.tsx:66]: Yes I planned to refactor the types, to have dedicated type for component and component open api, must I add this task to the backlog ?

File: src/state/entities/components/triggers.ts (hunk 2 above)
Comment (hkhattabii) [reply src/state/entities/components/triggers.ts:53]: No, I though that it is better to  display notifications when users create or update a component.

Comment (BerrevoetsRobbe) [reply src/state/entities/components/triggers.ts:53]: okay, could you also add this for scripts in a next issue? or add it to the backlog?

File: src/views/doc/OpenAPIOverview/index.tsx (hunk 3 above)
Comment (BerrevoetsRobbe) [reply src/views/doc/OpenAPIOverview/index.tsx:66]: yes, please add it to the backlog

**Review by BerrevoetsRobbe — APPROVED** [review]

**Comment by BerrevoetsRobbe** [comment]
please remove 'Load OpenApi documentation' from Scripts Overview
